memfs/memory.go

```diff
@@ -69,21 +75,40 @@ func (fs *Memory) OpenFile(filename string, flag int, perm os.FileMode) (billy.F
 	return f.Duplicate(filename, perm, flag), nil
 }
 
+func (fs *Memory) resolvePath(path string) string {
+	fullpath := clean(fs.Join(fs.base, path))
+	l, ok := fs.links[fullpath]
+	if !ok {
+		return fullpath
+	}
+
+	// IsAbs requires an unit so we check to if the path stars with a os.PathSeparator
+	if filepath.IsAbs(l.Target) || strings.HasPrefix(l.Target, string(os.PathSeparator)) {
+		return l.Target
+	}
+
+	return fs.resolvePath(fs.Join(filepath.Dir(fullpath), l.Target))
+}
+
 // Stat returns a billy.FileInfo with the information of the requested file.
 func (fs *Memory) Stat(filename string) (billy.FileInfo, error) {
-	fullpath := fs.Join(fs.base, filename)
-
+	fullpath := fs.resolvePath(filename)
 	f, has := fs.s.Get(fullpath)
 	if !has {
 		return nil, os.ErrNotExist
 	}
 
-	return f.Stat(), nil
+	// maybe this file belongs to a resolve link, so we use as name the
+	// filename of the opened path.
```

Comment:
In this case, you must to mention that something unexpected is happening in your implementation. `go-billy` is a replacement of a `os` package so, should mimic his behavior.